apps/server/src/modules/tool/external-tool/service/external-tool-configuration.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { EntityId, Page } from '@shared/domain';
+import { IToolFeatures, ToolFeatures } from '../../tool-config';
+import { ExternalTool } from '../domain';
+import { SchoolExternalTool } from '../../school-external-tool/domain';
+import { ContextExternalTool } from '../../context-external-tool/domain';
+import { CustomParameterScope } from '../../common/enum';
+import { CustomParameter } from '../../common/domain';
+import { ContextExternalToolTemplateInfo } from '../uc/dto';
+
+@Injectable()
+export class ExternalToolConfigurationService {
+	constructor(@Inject(ToolFeatures) private readonly toolFeatures: IToolFeatures) {}
+
+	public filterForAvailableTools(externalTools: Page<ExternalTool>, toolIdsInUse: EntityId[]): ExternalTool[] {
+		const availableTools: ExternalTool[] = externalTools.data.filter(
```

Comment:
Maybe you could filter here in two steps:
```suggestion
		const visibleTools: ExternalTool[] = externalTools.data.filter(
			(tool: ExternalTool): boolean => !tool.isHidden)
		);
	       const availableTools: ExternalTool[] = visibleTools.data.filter(
			(tool: ExternalTool): boolean => !!tool.id && !toolIdsInUse.includes(tool.id)
		);
```
something like that